packages/swing-store/src/transcriptStore.js

```diff
@@ -608,6 +609,29 @@ export function makeTranscriptStore(
   `);
   sqlCountPopulatedSpanItems.pluck();
 
+  function repairTranscriptSpanRecord(metadata) {
+    const { vatID, startPos, endPos, hash, isCurrent, incarnation } = metadata;
+    const existing = sqlGetSpanMetadataFor.get(vatID, startPos, endPos);
+    if (existing) {
+      if (
+        !!existing.isCurrent !== !!isCurrent ||
+        existing.hash !== hash ||
+        existing.incarnation !== incarnation
+      ) {
+        throw Fail`repairTranscriptSpanRecord metadata mismatch: ${existing} vs ${metadata}`;
+      }
+    } else {
+      sqlWriteSpan.run(
```

Comment:
Yeah here again we're relying on the `(vatID, startPos)` uniqueness. I'd much prefer if the get metadata query returned the `endPos` and we manually checked consistency to return a meaningful error.

Reply:
addressed in 4020cac69a004ea274b46158e83675b8e59fd0f7